application/src/main/scala/com/azavea/franklin/database/StacItemDao.scala

```diff
@@ -141,16 +149,33 @@ object StacItemDao extends Dao[StacItem] {
     Update[StacItemBulkImport](insertFragment).updateMany(stacItemInserts)
   }
 
-  def insertStacItem(item: StacItem): ConnectionIO[StacItem] = {
+  def insertStacItem(item: StacItem): ConnectionIO[Either[CollectionNotFound.type, StacItem]] = {
```

Comment:
`CollectionNotFound.type` instead of `StacItemDaoError` because it can only return that one, and forcing downstream consumers to handle any member of the type is worse.